test/lib/utils/joinHelper.ts

```diff
@@ -0,0 +1,74 @@
+import { replaceWrapped, Token } from '../../../src';
+import { sendTransactionGetBalances } from './helper';
+import { expect } from 'vitest';
+import { JoinTxInput } from './types';
+
+export const doJoin = async (txInput: JoinTxInput) => {
+    const {
+        poolJoin,
+        poolInput,
+        joinInput,
+        testAddress,
+        client,
+        slippage,
+        checkNativeBalance,
+        chainId,
+    } = txInput;
+    const queryResult = await poolJoin.query(joinInput, poolInput);
+    const { call, to, value, maxAmountsIn, minBptOut } = poolJoin.buildCall({
+        ...queryResult,
+        slippage,
+        sender: testAddress,
+        recipient: testAddress,
+    });
+
+    const poolTokens = poolInput.tokens.map(
+        (t) => new Token(chainId, t.address, t.decimals),
+    );
+    const bptIndex = poolInput.tokens.findIndex(
+        (t) => t.address === poolInput.address,
+    );
+    let tokensForBalanceCheck; // Replace with native asset if required
+    const poolTokensAddr = checkNativeBalance
+        ? replaceWrapped(poolTokens, chainId).map((t) => t.address)
+        : poolTokens.map((t) => t.address);
+    if (bptIndex >= 0) {
```

Comment:
Also a bit of comment here and LN53 would be useful as an explainer of why this is being done.